### PR TITLE
Events Prefab/Class

### PR DESCRIPTION
**Pull request by 0xFirekeeper**
Prefab_Events
Supports Get and GetAll event functions
Filters seem to be ignored with Dictionary, block filter also throws Invalid blockTag

### CODE REVIEW

File: Assets/Thirdweb/Core/Scripts/Types.cs [hunk 1]
```diff
@@ -456,4 +456,65 @@ public struct FundWalletOptions
...
+        public int blockNumber;
+        public string blockHash;
+        public int transactionIndex;
+        public bool removed;
+        public string address;
+        public string data;
+        public List<string> topics;
+        public string transactionHash;
+        public int logIndex;
+        public string @event;
```
Comment (joaquim-verges) [inline Assets/Thirdweb/Core/Scripts/Types.cs:500]: is this right? looks weird

**Comment by 0xFirekeeper** [comment]
Prefab_Events (V1)
Fetch and manipulate Contract Events with a simple API!
- Get specific events from any contract
- Get all events from any contract
- Optional string filter

File: Assets/Thirdweb/Examples/Scripts/Prefab_Events.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+public class Prefab_Events : MonoBehaviour
+{
+
+    // Get all events filtered by name (and optionally add more filters)
+
+    public async void GetEventsTest()
+    {
+        Contract contract = new Contract("goerli", "0x2e01763fA0e15e07294D74B63cE4b526B321E389");
+
+        List<ContractEvent<TransferEvent>> allEvents = await contract.events.Get<TransferEvent>("Transfer");
```
Comment (joaquim-verges) [inline Assets/Thirdweb/Examples/Scripts/Prefab_Events.cs:33]: wait so passing `tokenId = 0` doesnt work here? (with the corrected event property name)

Comment (0xFirekeeper) [reply Assets/Thirdweb/Examples/Scripts/Prefab_Events.cs:33]: it does, you can still pass it as a string, problem is with a type that contains fromBlock, toBlock and order + the filter dictionary, only using one of these sets the rest to default, which if fromBlock and toBlock is not set, they're 0

Sticking to a string solves this for now while I work on a different API with those filters as separate parameters to the direct Get and GetAll calls

**Comment by 0xFirekeeper** [comment]
Ready version with EventQueryOptions type as the parameter

**Review by joaquim-verges — APPROVED** [review]
nice! looks good

**Pull request merged**